hedera-node/hedera-mono-service/src/main/java/com/hedera/services/store/contracts/AbstractLedgerWorldUpdater.java

```diff
@@ -98,13 +99,18 @@
     protected int thisRecordSourceId = UNKNOWN_RECORD_SOURCE_ID;
     protected List<Integer> committedRecordSourceIds = Collections.emptyList();
     protected RecordsHistorian recordsHistorian = null;
+    protected AutoCreationLogic autoCreationLogic = null;
 
     protected Set<Address> deletedAccounts = new HashSet<>();
     protected Map<Address, UpdateTrackingLedgerAccount<A>> updatedAccounts = new HashMap<>();
 
-    protected AbstractLedgerWorldUpdater(final W world, final WorldLedgers trackingLedgers) {
+    protected AbstractLedgerWorldUpdater(
+            final W world,
+            final WorldLedgers trackingLedgers,
+            final AutoCreationLogic autoCreationLogic) {
         this.world = world;
         this.trackingLedgers = trackingLedgers;
+        this.autoCreationLogic = autoCreationLogic;
```

Comment:
⚠️ We can't use the `AutoCreationLogic` singleton here because it [directly changes](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-mono-service/src/main/java/com/hedera/services/txns/crypto/AutoCreationLogic.java#L239) the "base reality" `AliasManager` (not a [`StackedContractAliases` instance](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-mono-service/src/main/java/com/hedera/services/ledger/accounts/StackedContractAliases.java#L31) that buffers changes for a single frame).

The main problem is that `revert()` no longer works for aliases. For example, if an auto-creation is committed in frame `N + 1` and creates alias `X`, then even if frame `N` reverts, **the [underlying aliases `FCHashMap`](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-mono-service/src/main/java/com/hedera/services/state/org/StateMetadata.java#L31) will still include `X`**. Even worse, since alias `X` will not have been written to the serialized state (i.e., the accounts `MerkleMap`), a node that restarts will no longer believe `X` exists; and will likely ISS given any transaction that references it.

A related problem is that if an unhandled exception happens after an auto-creation succeeds, we will fail to remove its created aliases from the singleton `AliasManager`.

Reply:
Yikes... that would not have been pretty 😨 

Added method for construction of `AutoCreationLogic` in the `InfrastructureFactory` for the precompiles when needed.